### DISCUSSION TITLE
New install of TeamPass 3.1.1

### DISCUSSION THREAD

**Discussion by thifresan** [nilsteampassnet/TeamPass#4037] [Q&A] [answered]
When installing the new version of TeamPass 3.1.1 I receive the error shown in the image.

![WINWORD_VLKTr4ZH9s](https://github.com/nilsteampassnet/TeamPass/assets/113436180/89d56529-d3c9-4c84-9f5b-3b03ac860057)
![WINWORD_P2mZjN6xM1](https://github.com/nilsteampassnet/TeamPass/assets/113436180/d3b8bbd5-9308-4cb8-bcea-606480922253)

Web browsing log:

POST http://10.1.0.26/install/install.queries.php 500 (Internal Server Error)
send @ jquery.min.js:18
ajax @ jquery.min.js:18
doGetJson @ install.js:263
checkPage @ install.js:147
onclick @ install.php:111

Apache log:

[Wed Jan 31 18:56:00.110006 2024] [php:error] [pid 3129] [client 10.1.31.134:54032] PHP Fatal error:  Uncaught Error: Call to a member function has() on null in /var/www/html/teampass/includes/config/settings.php:28\nStack trace:\n#0 /var/www/html/teampass/sources/main.functions.php(4315): require_once()\n#1 /var/www/html/teampass/sources/main.functions.php(64): loadClasses()\n#2 /var/www/html/teampass/install/install.queries.php(51): require_once('...')\n#3 {main}\n  thrown in /var/www/html/teampass/includes/config/settings.php on line 28, referer: http://mydomain.com.br/install/install.php

**Comment by silfius** [comment]
I have more and les the same problem.
My Environment: 

- Debian 12
- Apache Server version: Apache/2.4.57 (Debian) / Server built:   2023-04-13T03:26:51
- PHP 8.2.7
- Mariadb 10.11.4-MariaDB-1~deb12u1 Debian 12


teampass_error.log
`[Fri Feb 02 12:36:37.813867 2024] [php:error] [pid 32020] [client 185.242.250.185:8794] PHP Fatal error:  Uncaught Error: Call to a member function has() on null in /var/www/html/TeamPass/includes/config/settings.php:28\nStack trace:\n#0 /var/www/html/TeamPass/sources/main.functions.php(4315): require_once()\n#1 /var/www/html/TeamPass/sources/main.functions.php(64): loadClasses()\n#2 /var/www/html/TeamPass/install/install.queries.php(51): require_once('...')\n#3 {main}\n  thrown in /var/www/html/TeamPass/includes/config/settings.php on line 28, referer: http://local.server.com/install/install.php
`
![image](https://github.com/nilsteampassnet/TeamPass/assets/73602363/fe9b6928-0a37-4778-bb39-32ee6f47d5ef)
![image](https://github.com/nilsteampassnet/TeamPass/assets/73602363/49724795-076d-441b-be1e-59c616018c0e)

**Comment by thifresan** [comment]
The issue occurs when the following permissions are granted:

chown -R www-data:www-data /var/www/html/teampass/ 
chmod -R 775 /var/www/html/teampass/

 chmod -R 0777 /var/www/html/teampass/includes/
 chmod -R 0777 /var/www/html/teampass/includes/config
 chmod -R 0777 /var/www/html/teampass/includes/avatars
 chmod -R 0777 /var/www/html/teampass/includes/libraries/csrfp/libs
 chmod -R 0777 /var/www/html/teampass/includes/libraries/csrfp/log
 chmod -R 0777 /var/www/html/teampass/includes/libraries/csrfp/js
 chmod -R 0777 /var/www/html/teampass/backups
 chmod -R 0777 /var/www/html/teampass/files
 chmod -R 0777 /var/www/html/teampass/install
 chmod -R 0777 /var/www/html/teampass/upload

**Reply by silfius** [reply]
I get with modify of settings.php
`//if (isset($_SESSION['settings']['timezone']) === true) {
 //   date_default_timezone_set($_SESSION['settings']['timezone']);
//}
`

**Reply by Arash9703** [reply, marked as answer]
Hello,
@silfius Is this the correct script to replace with the existing one in settings.php ? 
Thanks

**Reply by Arash9703** [reply]
@thifresan did it solve the problem for you !

**Reply by silfius** [reply]
Yes, but at the beginning, there are two settings.php.
You have to modify this one "/var/www/html/TeamPass/includes/config/settings.php"

**Reply by Arash9703** [reply]
https://github.com/nilsteampassnet/TeamPass/issues/4038#issuecomment-1944111295

**Reply by Arash9703** [reply]
I had forget to modify this one 🤦‍♂️ there are two settings.php
Pbm solved finally :)

**Comment by thifresan** [comment]
Hi @Arash9703 ! 

Sorry for the delay in responding to you. 

Thank you very much for your help my friend, it worked perfectly.
